### ISSUE TITLE
Deep Metric Learning with BIER: Boosting Independent Embeddings Robustly

### ISSUE THREAD

**Issue by sweaterr** [sweaterr/readinglist#392]

**Comment by sweaterr** [comment]
## 3.1 Online Gradient Boosting CNNs for Metric Learning

F(x (1) , x (2)) denotes the ensemble output and αm is the weighting factor of the m-th learner.
F (x (1), x (2))는 앙상블 출력을 나타내며, αm은 m 번째 학습자의 가중치이다.

The m-th learner of the ensemble is trained on a reweighted training batch according to the negative gradient −` 0 (·) of the loss function at the ensemble prediction until stage m − 1.
앙상블의 m 번째 학습자는 단계 m - 1까지의 앙상블 예측에서 손실 함수의 음의 기울기 -`0 (·)에 따라 재 학습 된 훈련 배치에 대해 학습된다.

**Comment by sweaterr** [comment]
To train the weak learners fm(·) in an online fashion, we adapt an online gradient boosting learning algorithm [44] with fixed weights αm and integrate it within a CNN.
약한 학습자 fm (·)을 온라인 방식으로 훈련시키기 위해 우리는 고정 가중치 αm을 갖는 온라인 그라디언트 부스팅 학습 알고리즘 [44]을 적용하여이를 CNN 내에 통합한다.

Na¨ıvely training multiple CNNs within the boosting framework is, however, computationally too expensive.
그러나 부스트 프레임 워크 내에서 여러 CNN을 교육하는 것은 계산 상 너무 비쌉니다.

To avoid this additional computational cost, we divide the embedding layer of our CNN into several non-overlapping groups, as illustrated in Fig  3.
이러한 추가적인 계산 비용을 피하기 위해 그림 3에서와 같이 CNN의 임베딩 레이어를 여러 비 중첩 그룹으로 나눕니다.

A single group represents a weak learner.
단일 그룹은 약한 학습자를 나타냅니다.

All our weak learners share the same underlying feature representation, which is a pre-trained ImageNet CNN in all our experiments.
모든 약한 학습자는 동일한 기본 피쳐 표현을 공유합니다.이 표현은 사전 실험 된 ImageNet CNN으로 모든 실험에서 동일합니다.

**Comment by sweaterr** [comment]
Our network is trained end-to-end on mini-batches with SGD and momentum.
SGD의 네트워크는 SGD 및 추진력을 갖춘 미니 배치 (mini-batches)에서 종단 간 (end-to-end)으로 교육됩니다.

We illustrate the training procedure for loss functions operating on pairs and a single example per batch in Algorithm 1.
우리는 쌍으로 동작하는 손실 함수에 대한 훈련 절차와 알고리즘 1에서 배치 당 하나의 예를 설명합니다.

Our algorithm also works with triplets, but for the sake of clarity we omit a detailed explanation here and refer the interested reader to the supplementary material, The training procedure can be easily integrated into the standard backpropagation algorithm, introducing only negligible additional cost, since most time during training is spent on computing convolutions.
우리의 알고리즘은 3 중항에도 적용되지만, 여기서는 자세한 설명을 생략하고 관심있는 독자를 보충 자료로 안내합니다. 훈련 절차는 표준 백 프로덕션 알고리즘에 쉽게 통합 될 수 있으며, 대부분의 경우 훈련 중 시간은 회선 계산에 소비됩니다.

First, in the forward pass we compute similarity scores s m n for each input sample n and each group m.
먼저, 순방향 패스에서 각 입력 샘플 n과 각 그룹 m에 대한 유사도 s m n을 계산합니다.

In the backward pass we backpropagate the reweighted losses for each group iteratively.
역방향 패스에서 우리는 반복적으로 각 그룹에 대해 재 손실 된 손실을 역 전파합니다.

The weight w m n for the n-th sample and the m-th learner is computed from the negative gradient −` 0 (·) of the ensemble prediction until stage m − 1.
n 번째 샘플과 m 번째 학습자에 대한 가중치 w m n은 단계 ​​m - 1까지의 앙상블 예측의 음의 기울기 -`0 (·)로부터 계산된다.

Hence, successive learners focus on examples which have large gradients (i.e, are misclassified) by previous learners.
따라서, 연속 학습자는 이전 학습자에 의해 큰 경사도 (즉, 잘못 분류 됨)를 갖는 예에 초점을 맞춘다.

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/41189375-d4daedb4-6c07-11e8-931a-51526f1c5201.png)
![image](https://user-images.githubusercontent.com/1518919/41189379-d8e4bfa2-6c07-11e8-863a-6dbba3b955f8.png)

**Comment by sweaterr** [comment]
## 3.2 Diversity Loss Functions

Rather than relying on boosting alone to increase the diversity in our ensemble, we propose additional loss functions which make learners more diverse from each other.
앙상블의 다양성을 높이기 위해 혼자 힘을 키우는 것에 의존하기보다는 학습자가 서로보다 다양 해지는 추가 손실 기능을 제안합니다.

We present two different loss functions to encourage the diversity of learners.
우리는 학습자의 다양성을 장려하기 위해 두 가지 다른 손실 함수를 제시합니다.

These can either be used for weight initialization or as auxiliary loss function during training (see Section 3.3).
이것들은 무게 초기화 또는 훈련 중 보조 손실 기능으로 사용될 수 있습니다 (3.3 참조).

Our first loss function, which we denote as Activation Loss, optimizes the embeddings such that for a given sample, only a single embedding is active and all other embeddings are close to zero (see Section 3.2.1).
우리가 Activation Loss로 표시하는 첫 번째 손실 함수는 주어진 샘플에 대해 단일 임베드 만 활성화되고 다른 모든 임베딩은 0에 가깝도록 임베딩을 최적화합니다 (3.2.1 절 참조).

As second loss function, we propose an Adversarial Loss.
두 번째 손실 함수로, 우리는 적의 손실을 제안합니다.

We train a regressor on top of our embeddings which maps one embedding to a different embedding, maximizing their similarity.
하나의 임베딩을 다른 임베딩에 매핑하여 유사성을 극대화하는 임베딩 맨 위에 회귀선을 훈련시킵니다.

By inserting a gradient reversal layer [19] between the regressors and our embeddings, we update our embeddings so that they minimize the similarity between each other with respect to these regressors which results in more diverse embeddings (see Section 3.2.2).
회귀 변수와 우리의 삽입 사이에 그라디언트 역전 계층을 삽입함으로써, 우리는 이들 회귀 분석기에 대한 서로 간의 유사성을 최소화하여보다 다양한 임베딩을 유도 할 수 있도록 임베딩을 업데이트합니다 (3.2.2 절 참조).
